### PR TITLE
[PATCH v4] Final platform build system cleanups

### PR DESCRIPTION
**Pull request by lumag**
These are the last changes modifying/generifying/cleaning up platform code and build system.

### CODE REVIEW

**Comment by Bill-Fischofer-Linaro** [comment]
Checkpatch flags this:
```
WARNING: line over 80 characters
#34: FILE: example/l2fwd_simple/l2fwd_simple_run.sh:12:
+./odp_l2fwd_simple${EXEEXT} pcap:in=${PCAP_IN} pcap:out=pcapout.pcap 02:00:00:00:00:01 02:00:00:00:00:02 &
total: 0 errors, 1 warnings, 0 checks, 61 lines checked
NOTE: Ignored message types: AVOID_EXTERNS BIT_MACRO COMPARISON_TO_NULL DEPRECATED_VARIABLE NEW_TYPEDEFS PREFER_PRINTF PREFER_SCANF SPLIT_STRING SSCANF_TO_KSTRTO VOLATILE
0007-example-add-EXEEXT-when-calling-compiled-programs.patch has style problems, please review.
```
@muvarov Not clear why the checkbox summary above shows checkpatch success but checkpatch reported in the Travis run fails.

@lumag Travis also reports distcheck test fails with this PR.

**Comment by lumag** [comment]
Fixed `shmem_linux` issue.

File: example/l2fwd_simple/l2fwd_simple_run.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@
 PCAP_IN=`find . ${TEST_DIR} $(dirname $0) -name udp64.pcap -print -quit`
 echo "using PCAP_IN = ${PCAP_IN}"
 
-./odp_l2fwd_simple pcap:in=${PCAP_IN} pcap:out=pcapout.pcap 02:00:00:00:00:01 02:00:00:00:00:02 &
+./odp_l2fwd_simple${EXEEXT} pcap:in=${PCAP_IN} pcap:out=pcapout.pcap 02:00:00:00:00:01 02:00:00:00:00:02 &
```
Comment (Bill-Fischofer-Linaro) [inline example/l2fwd_simple/l2fwd_simple_run.sh]: Please change this to:
```
./odp_l2fwd_simple${EXEEXT} pcap:in=${PCAP_IN} pcap:out=pcapout.pcap \
		  02:00:00:00:00:01 02:00:00:00:00:02 &
```
To eliminate checkpatch error.

Comment (lumag) [reply example/l2fwd_simple/l2fwd_simple_run.sh]: done

File: m4/odp_openssl.m4 [hunk 2]
```diff
@@ -10,7 +10,7 @@ AC_ARG_VAR([OPENSSL_STATIC_LIBS], [static linker flags for OpenSSL crypto librar
 # Set optional OpenSSL path
 ##########################################################################
 AC_ARG_WITH([openssl-path],
-[AC_HELP_STRING([--with-openssl-path=DIR],
+[AS_HELP_STRING([--with-openssl-path=DIR],
```
[the remark this answers is not in the document]
Comment (lumag) [reply m4/odp_openssl.m4]: @muvarov added commit message

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**